tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +302,110 @@ def guess_realert_every(job):
     return realert_every
 
 
-def compute_check_result_for_job(client, job):
-    kwargs = {
-        "name": "check_tron_job.{}".format(job['name']),
-        "source": "tron",
+def filter_monitoring_config(config):
+    """ Filters out non-Sensu arguments from a job's monitoring config """
+    sensu_args = set([
+        'name',
+        'runbook',
+        'status',
+        'output',
+        'team',
+        'page',
+        'tip',
+        'notification_email',
+        'check_every',
+        'realert_every',
+        'alert_after',
+        'dependencies',
+        'irc_channels',
+        'slack_channels',
+        'ticket',
+        'project',
+        'source',
+        'tags',
+        'ttl',
+        'sensu_host',
+        'component',
+        'description',
+    ])
+
+    include, exclude = {}, {}
+    for k, v in config.items():
+        (include if k in sensu_args else exclude)[k] = v
+    return include, exclude
+
+
+def sort_runs_by_interval(job_content, interval='day'):
+    """ Sorts a job's runs by a time interval (year, month, day, hour, minute,
+    or second), according to a job run's run time.
+    """
+    interval_formats = {
+        'year': '%Y',
+        'month': '%Y.%m',
+        'day': '%Y.%m.%d',
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
-    if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
 
-    status = job["status"]
-    if status == "disabled":
-        kwargs["output"] = "OK: {} is disabled and won't be checked.".format(
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        # Sort runs by shortdate
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            shortdate = time.strftime(
```

Comment:
If you are going to use the name "shortdate" as the variable name, then please remove the indirection of `interval_formats`

I realize you used a shorter variable for testing, but in practice we are never going to do anything other than a `shortdate`.

Reply:
I'll change the `shortdate` to something else, but I'd like to keep `interval_formats`, since it makes testing easier, now and in the future.